src/Symfony/Component/Finder/Gitignore.php

```diff
@@ -14,120 +14,78 @@
 /**
  * Gitignore matches against text.
  *
+ * @author Michael Voříšek <vorismi3@fel.cvut.cz>
  * @author Ahmed Abdou <mail@ahmd.io>
  */
 class Gitignore
 {
     /**
      * Returns a regexp which is the equivalent of the gitignore pattern.
      *
-     * @return string The regexp
+     * Format specification: https://git-scm.com/docs/gitignore#_pattern_format
      */
     public static function toRegex(string $gitignoreFileContent): string
     {
-        $gitignoreFileContent = preg_replace('/^[^\\\r\n]*#.*/m', '', $gitignoreFileContent);
-        $gitignoreLines = preg_split('/\r\n|\r|\n/', $gitignoreFileContent);
+        $gitignoreFileContent = preg_replace('~(?<!\\\\)#[^\n\r]*~', '', $gitignoreFileContent);
+        $gitignoreLines = preg_split('~\r\n?|\n~', $gitignoreFileContent);
```

Comment:
I found `\r\n|\r|\n` more readable/maintainable

Reply:
No, `~` regex delimiter is now used here solely which is much better when working with paths as `/` does not need to be backslashed. Greatly improves readability.

Reply:
@mvorisek the comment was not about the delimiter